src/commands/ImportExportCommands.cpp

```diff
@@ -84,7 +84,7 @@ void ExportCommand::PopulateOrExchange(ShuttleGui & S)
 
    S.StartMultiColumn(2, wxALIGN_CENTER);
    {
-      S.TieTextBox(XXO("File Name:"),mFileName);
+      S.TieTextBox(XXO("File Name:"), mFileName, mFileName.length());
```

Comment:
this fix kinda works, but runs into some issues: 
1. the size of the window is dependent on the last exported file. For example, if I export D:\A.wav and open the dialog again, it's as short as ever:
![image](https://user-images.githubusercontent.com/87814144/158130400-d3c1f88f-90aa-4581-b74b-35228c81a398.png)
Whereas if I just had an AAA.....wav, it is somewhat longer: 
![image](https://user-images.githubusercontent.com/87814144/158130490-13ed9ac2-6ce9-411b-9527-338e54594b83.png)
2. using any sort of `string.length()` will give you a value that never works for variable-width font. III and WWW are both length 3, but one is obviously much wider than the other. 

I think a better fix for this issue could be either one of these: 

* A text box that's relative to the window size, and the window being expandable to whatever width you need it to be. 
* A "browse" button next to the text field that lets you select the file in the file browser, similar to what the normal import/export dialog windows would do. You may want to take a look at how the ffmpeg library locator did it, as it also can display a path. 
![image](https://user-images.githubusercontent.com/87814144/158132818-30537a09-d0cf-404a-82ce-35719be2e41e.png)

Reply:
Hey @LWinterberg ,

So I tried looking into the 'Browse' button solution but got stuck at a certain point. I was able to create a 'Browse' button at a location similar to the `Locate FFmpeg` window.

However, I am unsure how to connect the button to a window that allows you to select a file. I have looked into the FFmpeg class and have tried to mimic the logic there. However I think I do not understand the codebase to a good extent yet.

From what I understand you define an `EVENT_TABLE` and then link an event to some ID. At the same time the button is linked to that ID too. I basically followed what was done in the `FFmpeg` file. 
However I have trouble figuring the eventhandling part of the code. So I have the following:

![image](https://user-images.githubusercontent.com/23222744/159168315-a88dc106-eda0-430b-b764-a8a1b5a1f74a.png)

![image](https://user-images.githubusercontent.com/23222744/159168282-250d8057-8719-45ac-93ce-7666d50af6b0.png)

An `EVENT_TABLE` where I link the button to an `onBrowse` function that tries to mimic the same behaviour as what is defined in ``FFmpeg`. However when I try this I ran into some issues. 
- the `onBrowse` in `FFmpeg` uses a `FileSelector` and I can't seem to satisfy all the parameters required in the class `ImportExportCommands` which I am working in.

I also think that my approach could be entirely wrong due to my limited knowledge about the codebase. I was not really able to find any sources to understand things better. Maybe you could point me into the right direction or point me towards some sources I could take a look at?

Reply:
I'm unfortunately not a programmer myself, but maybe some of the following is helpful: 
1. There's a code documentation thing on https://doxy.audacityteam.org
2. if you install Graphviz (as described in the BUILDING.md), you'll get images showing you how various libraries depend on each other. 

Tell me whether that's been helpful, if not i can ask the team tomorrow

Reply:
Thanks. I think these will be very helpful. I'll let you know if I have any specific questions.